<?xml version="1.0"?>
<launch>
  
  <arg name="camera_frame_id" default="camera_optical_frame" />
  <arg name="rqt" default="false" />
  <arg name="rviz" default="false" />
  
  <!-- Logging -->
  <env name="ROSCONSOLE_CONFIG_FILE" value="$(find ensenso)/config/rosconsole.conf"/>
  
  <!-- Ensenso driver -->
  <node name="ensenso_driver" pkg="ensenso" type="ensenso_driver" output="screen"> 
    <!-- Static parameters -->
    <rosparam param="cameras">
        - name: sensor_front
          serial: 160678
        - name: sensor_rear
          serial: 171576
    </rosparam>
  </node>
  
  <group if="$(arg rqt)">
    <node name="rqt_gui_node" pkg="rqt_gui" type="rqt_gui" args="--perspective-file $(find ensenso)/config/viewer.perspective" />
  </group>
  
  <group if="$(arg rviz)">
    <node name="rviz_node" pkg="rviz" type="rviz" args="-d $(find ensenso)/config/viewer.rviz" />
  </group>

</launch>
